samza-api/src/main/java/org/apache/samza/storage/kv/KeyValueSnapshot.java

```diff
@@ -19,6 +19,22 @@
 
 package org.apache.samza.storage.kv;
 
-public interface KeyValueIterable<K, V> extends Iterable<Entry<K, V>> {
+/**
+ * An immutable view of the {@link KeyValueStore} at a point-in-time.
```

Comment:
Can we add a similar comment to the iterator method that this snapshot MUST be closed after use.

Reply:
done.